crypto/lhash/lhash.c

```diff
@@ -44,26 +44,42 @@ static int expand(OPENSSL_LHASH *lh);
 static void contract(OPENSSL_LHASH *lh);
 static OPENSSL_LH_NODE **getrn(OPENSSL_LHASH *lh, const void *data, unsigned long *rhash);
 
+OPENSSL_LHASH *OPENSSL_LH_set_thunks(OPENSSL_LHASH *lh,
+                                     OPENSSL_LH_HASHFUNCTHUNK hw,
+                                     OPENSSL_LH_COMPFUNCTHUNK cw,
+                                     OPENSSL_LH_DOALL_FUNC_THUNK daw,
+                                     OPENSSL_LH_DOALL_FUNCARG_THUNK daaw)
+{
+
+    if (lh == NULL)
+        return NULL;
+    lh->compw = cw;
+    lh->hashw = hw;
+    lh->daw = daw;
+    lh->daaw = daaw;
+    return lh;
+}
+
 OPENSSL_LHASH *OPENSSL_LH_new(OPENSSL_LH_HASHFUNC h, OPENSSL_LH_COMPFUNC c)
 {
-    OPENSSL_LHASH *ret;
+    struct lhash_st *new;
 
-    if ((ret = OPENSSL_zalloc(sizeof(*ret))) == NULL)
+    if ((new = OPENSSL_zalloc(sizeof(*new))) == NULL)
         return NULL;
-    if ((ret->b = OPENSSL_zalloc(sizeof(*ret->b) * MIN_NODES)) == NULL)
+    if ((new->b = OPENSSL_zalloc(sizeof(*new->b) * MIN_NODES)) == NULL)
         goto err;
-    ret->comp = ((c == NULL) ? (OPENSSL_LH_COMPFUNC)strcmp : c);
-    ret->hash = ((h == NULL) ? (OPENSSL_LH_HASHFUNC)OPENSSL_LH_strhash : h);
-    ret->num_nodes = MIN_NODES / 2;
-    ret->num_alloc_nodes = MIN_NODES;
-    ret->pmax = MIN_NODES / 2;
-    ret->up_load = UP_LOAD;
-    ret->down_load = DOWN_LOAD;
-    return ret;
+    new->comp = ((c == NULL) ? (OPENSSL_LH_COMPFUNC)strcmp : c);
+    new->hash = ((h == NULL) ? (OPENSSL_LH_HASHFUNC)OPENSSL_LH_strhash : h);
+    new->num_nodes = MIN_NODES / 2;
+    new->num_alloc_nodes = MIN_NODES;
+    new->pmax = MIN_NODES / 2;
+    new->up_load = UP_LOAD;
+    new->down_load = DOWN_LOAD;
+    return new;
 
 err:
-    OPENSSL_free(ret->b);
-    OPENSSL_free(ret);
+    OPENSSL_free(new->b);
+    OPENSSL_free(new);
     return NULL;
 }
```

Comment:
Might I request that cosmetic changes (which is what this is is) are made in a separate commit?

Reply:
Yes, but these  are a holdover from when I extended the lhash structure.  I can just revert this to use ret again instead of new

Reply:
reverted to its initial form